### ISSUE TITLE
Use expecttest in test_compiled_optimizers.py

### ISSUE THREAD

**Issue by mlazos** [pytorch/pytorch#141262]
### 🐛 Describe the bug

@ezyang pointed out that it's kind of annoying to update the kernel counts in https://github.com/pytorch/pytorch/blob/723498aab8803702dfd508be05a1ebe525112ebd/test/inductor/test_compiled_optimizers.py#L4

We should update this to use expecttest like our other tests.

### Versions

N/A

**Comment by tajusa14** [comment]
I'll have a look at this

**Comment by ezyang** [comment]
wait for https://github.com/pytorch/pytorch/pull/138922 to land first

**Comment by mlazos** [comment]
@tajusa14 that PR has landed, you should be good to go

**Comment by enCCzu** [comment]
Hello! I noticed that @tajusa14 is looking into this. I would love to contribute as well! Would you mind if I gave it a try or collaborated?

**Comment by enCCzu** [comment]
Hello again! From the contribution guidelines, I understand that multiple people can work on an issue, but let me know if I am intruding!
I wanted to share my understanding of the issue and ask for some guidance (anyone's is much appreciated) to ensure I’m on the right track: 
1. Using `expecttest` should help eliminate the reliance on hardcoded optimizer values in `KERNEL_COUNT_OVERRIDES`, `KERNEL_COUNTS`, and potentially parts of `CompiledOptimizerTests`. Is this correct?
2. Should this fix aim to include all supported optimizers dynamically (such as by introspecting `torch.optim`) or focus only on the current list of (what I believe to be) popular optimizers in `KERNEL_COUNTS`?
3. `expecttest` automatically populates the expected output in triple quotes (`""" """`) after running the test. Just to confirm, the fix for this issue should include these populated triple quotes, similar to tests in files like `test/profiler/test_profiler.py`, correct?

I'll continue working on a solution in the meantime. Thank you so much for your time, and I hope these questions weren't too rudimentary/obvious.

**Comment by enCCzu** [comment]
Understood! Thanks for the heads up. Also, thank you @mlazos for your guidance earlier! I was able to understand the issue much better and made good progress.

**Comment by tajusa14** [comment]
Alright, please tell us when the code has relanded

**Comment by nirajkamal** [comment]
Hi, I am trying to contribute for the first time to pytorch - is this issue still active!? Please let me know.

**Comment by mlazos** [comment]
> Hi, I am trying to contribute for the first time to pytorch - is this issue still active!? Please let me know.

Yes I don't think this has landed, please look at the above context and let me know if you have questions.
